### PR TITLE
Ug 987 rename save to list variant

### PR DESCRIPTION
**Pull request by leannecornish-ft**
This PR is to remove the old styling for the save-article-to-list from the code base - it hasn't been used since the feature flag for the new variant was turned on following the DAC audit. Specifically, the PR:

- Removes the old code
- Moves all the variant code into the main lists file (where the old code lived)
- Removes the temporary events that were configured for our tests and are no longer used
- Removes code that checks for the feature flag
- Remove any names that indicate variant rather than normal experience

** Testing **

1. Clone the branch locally, install and run npm link
2. Run next-article locally and link to this code, i.e. run npm link @financial-times/n-myft-ui (may need to use --force)
3. Regression test next-article by comparing the save article's functionality on local and prod, including:
              - saving an article to a new list
              - cancelling at various points
              - toggling manageArticleLists on and off
              - styling
              - desktop and mobile
              - Various browsers
4. Either do the same on an app that uses the modal version (e.g. next-myft-page), or pass in `modal: true` [here](https://github.com/Financial-Times/next-article/blob/94e32bf5636c3091acb62a596d63477b293bea7d/pages/shared-components/main-share.jsx#L58) (locally). And run the same tests as step 3.

### CODE REVIEW

File: myft/ui/lists.js [hunk 1]
```diff
@@ -1,330 +1,516 @@
...
-			return openCreateListAndAddArticleOverlay(contentId, config);
+		const configSet = event.currentTarget.querySelector('[data-myft-ui-save-config]');
+		if (!configSet) {
+			return openCreateListAndAddArticleOverlay(contentId);
 		}
-
-		handleArticleSaved(contentId);
+		const configKeys = configSet ? configSet.dataset.myftUiSaveConfig.split(',') : [];
+		const config = configKeys.reduce((configObj, key) => (key ? { ...configObj, [key]: true} : configObj), {});
+		if (config['modeless']) {
```
[the remark this answers is not in the document]
Comment (leannecornish-ft) [reply myft/ui/lists.js:503]: I like that, thank you 🙌

File: myft/ui/lists.js [hunk 2]
```diff
@@ -1,330 +1,516 @@
...
-			}));
-
-			return openCreateListAndAddArticleOverlay(contentId, config);
+		const configSet = event.currentTarget.querySelector('[data-myft-ui-save-config]');
+		if (!configSet) {
+			return openCreateListAndAddArticleOverlay(contentId);
 		}
-
-		handleArticleSaved(contentId);
+		const configKeys = configSet ? configSet.dataset.myftUiSaveConfig.split(',') : [];
```
[the remark this answers is not in the document]
Comment (leannecornish-ft) [reply myft/ui/lists.js:501]: 👍 I thought you were being extra cautious 😅

File: components/save-for-later/save-for-later.html [hunk 3]
```diff
@@ -4,8 +4,7 @@
 	data-myft-ui="saved"
 	action="/myft/save/{{contentId}}"
 	data-js-action="/__myft/api/core/saved/content/{{contentId}}?method=put"
-	{{#if @root.flags.manageArticleLists}}data-myft-ui-save-new="manageArticleLists"{{/if}}
-	{{#if @root.flags.manageArticleLists}}data-myft-ui-save-new-config="{{#if modal}}modal{{/if}}"{{/if}}>
+	data-myft-ui-save-config="{{#if modal}}modal{{/if}}{{#if modeless}}modeless{{/if}}">
```
[the remark this answers is not in the document]
Comment (jamesnicholls) [reply components/save-for-later/save-for-later.html]: Can I suggest the word `non-modal`?

[the remark this answers is not in the document]
Comment (leannecornish-ft) [reply components/save-for-later/save-for-later.html]: Is that because it's a more well known term?

[the remark this answers is not in the document]
Comment (leannecornish-ft) [reply components/save-for-later/save-for-later.html]: I'm not sure about having a negative (i.e., 'non') = true. That feels confusing.

[the remark this answers is not in the document]
Comment (jamesnicholls) [reply components/save-for-later/save-for-later.html]: Yeah, the double-negative is confusing, but I wouldn't know what the term `modeless` means without additional context, although I understand the terms `modal` and `non-modal`.

**Pull request merged**